tests/CMakeLists.txt

```diff
@@ -53,10 +53,14 @@ set(TEST_SOURCES
 	"${CMAKE_SOURCE_DIR}/rwgame/GameInput.cpp"
 	)
 
-ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
-
 add_executable(run_tests ${TEST_SOURCES})
 
+if(NOT LINK_STATIC)
+	target_compile_definitions(run_tests
+		PUBLIC
+			BOOST_TEST_DYN_LINK)
+endif()
+
```

Comment:
I will change it to this: https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html